core/src/test/java/org/apache/calcite/test/SqlToRelTestBase.java

```diff
@@ -603,6 +609,7 @@ protected TesterImpl(DiffRepository diffRepos, boolean enableDecorrelate,
       this.plannerFactory = Objects.requireNonNull(plannerFactory, "plannerFactory");
       this.conformance = Objects.requireNonNull(conformance, "conformance");
       this.contextTransform = Objects.requireNonNull(contextTransform, "contextTransform");
+      this.typeFactory = typeFactory;
```

Comment:
I would create a overloaded version of this constructor so that the below codes remains unchanged

Reply:
Thanks for the comment. But I don't catch an idea. The methods `with<OptionName>` creates new instance of the `TesterImpl`. How we can use old constructor here without specified `typeFactory` and keep the `typeFactory` from current instance?

Reply:
is there a default typeFactory you can pass in at constructor ? I guess most of the places are referring to the same type factory?

Reply:
I don't see many places usage of the `TesterImpl` just (`RelOptRulesTest`, `SqlToRelConverterTest` and `SqlToRelTestBase`).
What do you propose to make it possible to customize `typeFactory` for the test?

Reply:
ok